sources/api/schnauzer/src/helpers.rs

```diff
@@ -1266,6 +1370,24 @@ fn pause_registry<S: AsRef<str>>(region: S) -> String {
     }
 }
 
+/// Constructs the fully qualified domain name for the TUF repository for the
+/// given region. Returns a default if the region is not mapped.
+fn tuf_repository<S: AsRef<str>>(region: S) -> String {
+    // lookup the repository bucket or fallback to the public url
+    let (region, bucket) = match TUF_BUCKET_MAP.borrow().get(region.as_ref()) {
+        None => (return TUF_PUBLIC_REPOSITORY.to_string()),
+        Some(bucket) => (region.as_ref(), *bucket),
+    };
+    let partition = match ALT_PARTITION_MAP.borrow().get(region) {
+        None => (STANDARD_PARTITION),
+        Some(partition) => *partition,
+    };
+    match partition {
+        "aws-cn" => format!("https://{}.s3.{}.amazonaws.com.cn/latest", bucket, region),
+        _ => format!("https://{}.s3.{}.amazonaws.com/latest", bucket, region),
+    }
```

Comment:
Per discussion, we should try to use the dual stack endpoints here for forward compatibility with IPv6-only subnets.

Reply:
Rather than just specifying the bucket, I appended `.s3.dualstack` to the strings directly _(and verified it still works for cn-north-1 and cn-northwest-1)_. In the event that we add a region that does not support dual stack endpoints we can just do `<bucket_id>.s3`. We could get fancier with the map, but I wanted to keep things simple.